app/controllers/local_authorities_controller.rb

```diff
@@ -3,6 +3,8 @@ class LocalAuthoritiesController < ApplicationController
   include LinkImporterUtils
 
   before_action :set_authority, except: %i[index bad_homepage_url_and_status_csv]
+  before_action :redirect_unless_gds_editor, except: %i[update download_links_csv upload_links_csv]
+  before_action :forbid_unless_gds_editor, only: %i[update download_links_csv upload_links_csv]
```

Comment:
What I've done is just been explicit about everything. There's an argument that we should do:
```
before_action blah_permission, only: [ etc]
before_action blah_permission 2, only: [ etc_a etc_b]
before_action blah_permission 3, except [listed above]
```

...because it means that if someone adds an action it always gets the final permission. But we don't add actions that frequently, so it might be better to only use `only` and always just list out all actions they apply to.

(I've also fixed the download to make it forbidden)

Reply:
I think perhaps this was made more confusing by having "unless" in the action name. Things being truthy can make the logic easier to follow.
i.e. rather than `forbid_unless_gds_editor` perhaps it could be named `allow_if_gds_editor`. Then even if an `except` was used there'd only be one negative in the logic. I think it was the double negatives that were causing the brain twisting.